syntax = "proto3";
package cluster;





/*
List请求
*/
message ListRequest {
    /*
    页码
    */
    int32 page = 1 ;
    /*
    每页大小
    */
    int32 page_size = 2 ;
}

/*
List返回
*/
message ListResponse {
message List {
message NamespaceList {
    /*
    namespace Id
    */
    string instanceId = 1 ;
    /*
    名称
    */
    string resourceName = 2 ;
}
        
message ResourceGroupList {
    /*
    namespace Id
    */
    string instanceId = 1 ;
    /*
    名称
    */
    string name = 2 ;
}
        
message AuthInfo {
    /*
    用户名
    */
    string username = 1 ;
    /*
    密码
    */
    string password = 2 ;
}
        
message TlsConfig {
    /*
    客户端证书
    */
    string clientCertificate = 1 ;
    /*
    客户端Key
    */
    string clientKey = 2 ;
    /*
    集群证书
    */
    string clusterCA = 3 ;
}
        
    
    /*
    命名空间列表
    */
    repeated NamespaceList namespaceList = 1 ;
    /*
    部署资源组列表
    */
    repeated ResourceGroupList resourceGroupList = 2 ;
    /*
    cluster id，服务端自动生成
    */
    string instanceId = 3 ;
    /*
    集群名称
    */
    string name = 4 ;
    /*
    集群地址
    */
    string host = 5 ;
    /*
    环境类型
    */
    string environment = 6 ;
    /*
    授权模式
    */
    string authMode = 7 ;
    /*
    管理账号
    */
    AuthInfo authInfo = 8 ;
    /*
    证书信息
    */
    TlsConfig tlsConfig = 9 ;
    /*
    默认 StorageClass
    */
    string defaultStorageClass = 10 ;
    /*
    备注
    */
    string description = 11 ;
    /*
    来源
    */
    string provider = 12 ;
    /*
    区域
    */
    string zone = 13 ;
    /*
    区域ID
    */
    string zoneId = 14 ;
    /*
    云服务提供商
    */
    string cloudProvider = 15 ;
    /*
    创建时间,服务端自动生成
    */
    string ctime = 16 ;
}
        
    
    /*
    页数
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated List list = 4 ;
}

/*
ListApi返回
*/
message ListResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    cluster.ListResponse data = 4 ;
}

